<launch>
    <!-- republish topic from compressed image -->
    <node name="republish_color" type="republish" pkg="image_transport" args="compressed in:=/camera/color/image_raw raw out:=/color" />
    <node name="republish_depth" type="republish" pkg="image_transport" args="compressed in:=/camera/aligned_depth_to_color/image_raw raw out:=/aligned_depth" />
		
    <node name="orb_slam3_rbgd" pkg="orb_slam3" type="orb_slam3_rgbd" output="screen">
        <!-- for Realsense D435i -->
        <remap from="/camera/rgb/image_raw"                 to="/color"/>
        <remap from="/camera/depth_registered/image_raw"    to="/aligned_depth"/>

        <!-- Parameters for original ORB-SLAM3 -->
        <param name="voc_file"      type="string"   value="$(find orb_slam3)/config/ORBvoc.bin" />
        <param name="settings_file" type="string"   value="$(find orb_slam3)/config/D435i_rgb_d_inertial.yaml" />
    
        <!-- Parameters for ROS -->
        <param name="publish_tf_transform"   type="bool"     value="false" />
        <param name="map_frame_id"  type="string"   value="world" />
        <param name="pose_frame_id" type="string"   value="camera_link" />
    </node>
    
    <!-- Visualization - RViz-->
    <node name="rviz" pkg="rviz" type="rviz" args="-d $(find orb_slam3)/config/orb_slam3.rviz" output="screen" />

    <node pkg="hector_trajectory_server" type="hector_trajectory_server" name="trajectory_server_orb_slam3" output="screen" ns="hector_trajectory_server" >
        <param name="/target_frame_name" value="/world" />
        <param name="/source_frame_name" value="/camera_link" />
        <param name="/trajectory_update_rate" value="20.0" />
        <param name="/trajectory_publish_rate" value="20.0" />
    </node>
</launch>
